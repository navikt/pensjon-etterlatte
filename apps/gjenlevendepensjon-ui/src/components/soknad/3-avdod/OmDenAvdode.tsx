import { SkjemaGruppe } from "../../felles/SkjemaGruppe";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { IAvdoed } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { useTranslation } from "react-i18next";
import { FormProvider, useForm } from "react-hook-form";
import { RHFFoedselsnummerInput, RHFInput, RHFValutaInput } from "../../felles/rhf/RHFInput";
import { RHFSpoersmaalRadio } from "../../felles/rhf/RHFRadio";
import { IValg } from "../../../typer/Spoersmaal";
import Feilmeldinger from "../../felles/Feilmeldinger";
import BoddEllerArbeidetUtland from "./fragmenter/BoddEllerArbeidetUtland";
import Navigasjon from "../../felles/Navigasjon";
import { BodyLong, Label, Heading, Cell, Grid } from "@navikt/ds-react";
import HvorforSpoerVi from "../../felles/HvorforSpoerVi";
import { deepCopy } from "../../../utils/deepCopy";
import { RHFSelect } from "../../felles/rhf/RHFSelect";
import { useLand } from "../../../hooks/useLand";
import {SkjemaGruppeIngress, SkjemaGruppeRad} from "../../felles/StyledComponents";
import {SkjemaElement} from "../../felles/SkjemaElement";
import Bredde from "../../../typer/bredde";

const OmDenAvdode: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();
    const { land } : { land: any } = useLand();
    const methods = useForm<IAvdoed>({
        defaultValues: { ...state.omDenAvdoede, statsborgerskap: state.omDenAvdoede.statsborgerskap },
        shouldUnregister: true,
    });

    const {
        handleSubmit,
        watch,
        getValues,
        formState: { errors },
    } = methods;

    const lagreNeste = (data: IAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_AVDOED, payload: {...deepCopy(data), erValidert: true} });
        neste!!();
    };

    const lagreTilbake = (data: IAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_AVDOED, payload: {...deepCopy(data), erValidert: true} })
        forrige!!()
    }

    const lagreTilbakeUtenValidering = () => {
        const verdier = getValues()
        dispatch({ type: ActionTypes.OPPDATER_AVDOED, payload: {...deepCopy(verdier), erValidert: false} })
        forrige!!()
    }

    const erValidert = state.omDenAvdoede.erValidert;
    const selvstendigNaeringsdrivende = watch("selvstendigNaeringsdrivende.svar");
    const harAvtjentMilitaerTjeneste = watch("harAvtjentMilitaerTjeneste.svar");

    return (
        <FormProvider {...methods}>
            <SkjemaElement>
                <Heading className={"center"} size={"medium"}>{t("omDenAvdoede.tittel")}</Heading>
            </SkjemaElement>

            <form>
                <SkjemaGruppe>
                    <SkjemaGruppeRad>
                        <div className={"kol"}>
                            <Label>{t("omDenAvdoede.fornavn")}</Label>
                            <BodyLong>{state.omDegOgAvdoed.avdoed?.fornavn || ""}</BodyLong>
                        </div>
                        <div className={"kol"}>
                            <Label>{t("omDenAvdoede.etternavn")}</Label>
                            <BodyLong>{state.omDegOgAvdoed.avdoed?.etternavn || ""}</BodyLong>
                        </div>
                    </SkjemaGruppeRad>

                    <SkjemaElement>
                        <Grid style={{alignItems: "end" }}>
                            <Cell xs={12} md={6} className={"kol"}>
                                <RHFFoedselsnummerInput
                                    name={"foedselsnummer"}
                                    label={t("omDenAvdoede.foedselsnummer")}
                                    placeholder={t("felles.fnrPlaceholder")}
                                />
                            </Cell>

                            <Cell xs={12} md={6} className={"kol"}>
                                <RHFSelect
                                    name={`statsborgerskap`}
                                    label={t("omDenAvdoede.statsborgerskap")}
                                    selectOptions={land}
                                />
                            </Cell>
                        </Grid>
                    </SkjemaElement>
                </SkjemaGruppe>

                <BoddEllerArbeidetUtland datoForDoedsfallet={state.omDegOgAvdoed.avdoed?.datoForDoedsfallet}/>

                <SkjemaGruppe>
                    <SkjemaGruppeIngress>
                        <Heading size="small">{t("omDenAvdoede.selvstendigNaeringsdrivende.tittel")}</Heading>
                        <BodyLong>{t("omDenAvdoede.selvstendigNaeringsdrivende.ingress")}</BodyLong>
                    </SkjemaGruppeIngress>

                    <SkjemaElement>
                        <RHFSpoersmaalRadio
                            name={"selvstendigNaeringsdrivende.svar"}
                            legend={t("omDenAvdoede.selvstendigNaeringsdrivende.svar")}
                            vetIkke
                        />
                    </SkjemaElement>

                    {selvstendigNaeringsdrivende === IValg.JA && (
                        <>
                            <SkjemaGruppe>
                                <RHFValutaInput
                                    name={"selvstendigNaeringsdrivende.beskrivelse"}
                                    htmlSize={Bredde.S}
                                    valgfri
                                    type="tel"
                                    placeholder={t("omDenAvdoede.selvstendigNaeringsdrivende.placeholder")}
                                    label={t("omDenAvdoede.selvstendigNaeringsdrivende.beskrivelse")}
                                />
                            </SkjemaGruppe>

                            <RHFSpoersmaalRadio
                                name={"haddePensjonsgivendeInntekt.svar"}
                                legend={t("omDenAvdoede.haddePensjonsgivendeInntekt.svar")}
                                vetIkke
                            />
                        </>
                    )}
                </SkjemaGruppe>

                <SkjemaGruppe>
                    <SkjemaGruppeIngress>
                        <Heading size="small">{t("omDenAvdoede.annenOpptjening.tittel")}</Heading>
                    </SkjemaGruppeIngress>

                    <SkjemaGruppe>
                        <RHFSpoersmaalRadio
                            name={"doedsfallAarsak"}
                            legend={t("omDenAvdoede.doedsfallAarsak")}
                            description={
                                <HvorforSpoerVi title="omDenAvdoede.doedsfallAarsak">{t("omDenAvdoede.doedsfallAarsakHvorfor")}</HvorforSpoerVi>}
                            vetIkke
                        />
                    </SkjemaGruppe>

                        <RHFSpoersmaalRadio
                            name={"harAvtjentMilitaerTjeneste.svar"}
                            legend={t("omDenAvdoede.harAvtjentMilitaerTjeneste.svar")}
                            description={
                                <HvorforSpoerVi title="omDenAvdoede.harAvtjentMilitaerTjeneste.svar">{t("omDenAvdoede.harAvtjentMilitaerTjeneste.hjelpetekst")}</HvorforSpoerVi>
                            }
                            vetIkke
                        />

                        {harAvtjentMilitaerTjeneste === IValg.JA && (
                            <SkjemaElement>
                                <RHFInput
                                    name={"harAvtjentMilitaerTjeneste.beskrivelse"}
                                    htmlSize={Bredde.S}
                                    valgfri
                                    label={t("omDenAvdoede.harAvtjentMilitaerTjeneste.beskrivelse")}
                                />
                            </SkjemaElement>
                        )}

                </SkjemaGruppe>

                <Feilmeldinger errors={errors}/>

                <Navigasjon
                    forrige={{ onClick: erValidert === true ? handleSubmit(lagreTilbake) : lagreTilbakeUtenValidering }}
                    neste={{ onClick: handleSubmit(lagreNeste) }}
                />

            </form>
        </FormProvider>
    );
};

export default OmDenAvdode;
