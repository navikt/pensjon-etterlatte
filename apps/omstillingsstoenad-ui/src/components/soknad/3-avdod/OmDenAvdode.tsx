import { SkjemaGruppe } from '../../felles/SkjemaGruppe'
import SoknadSteg from '../../../typer/SoknadSteg'
import { useSoknadContext } from '../../../context/soknad/SoknadContext'
import { IAvdoed } from '../../../typer/person'
import { ActionTypes } from '../../../context/soknad/soknad'
import { useTranslation } from 'react-i18next'
import { FormProvider, useForm } from 'react-hook-form'
import { RHFFoedselsnummerInput, RHFInput, RHFValutaInput } from '../../felles/rhf/RHFInput'
import { RHFSpoersmaalRadio } from '../../felles/rhf/RHFRadio'
import { IValg } from '../../../typer/Spoersmaal'
import Feilmeldinger from '../../felles/Feilmeldinger'
import BoddEllerArbeidetUtland from './fragmenter/BoddEllerArbeidetUtland'
import Navigasjon from '../../felles/Navigasjon'
import { BodyLong, Heading, HGrid } from '@navikt/ds-react'
import { deepCopy } from '../../../utils/deepCopy'
import { RHFSelect } from '../../felles/rhf/RHFSelect'
import { useLand } from '../../../hooks/useLand'
import { SkjemaElement } from '../../felles/SkjemaElement'
import Bredde from '../../../typer/bredde'
import Datovelger from '../../felles/Datovelger'

const OmDenAvdode: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation()
    const { state, dispatch } = useSoknadContext()
    const { land }: { land: any } = useLand()
    const methods = useForm<IAvdoed>({
        defaultValues: { ...state.omDenAvdoede, statsborgerskap: state.omDenAvdoede.statsborgerskap },
        shouldUnregister: true,
    })

    const {
        handleSubmit,
        watch,
        getValues,
        formState: { errors },
    } = methods

    const lagreNeste = (data: IAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_AVDOED, payload: { ...deepCopy(data), erValidert: true } })
        neste!!()
    }

    const lagreTilbake = (data: IAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_AVDOED, payload: { ...deepCopy(data), erValidert: true } })
        forrige!!()
    }

    const lagreTilbakeUtenValidering = () => {
        const verdier = getValues()
        dispatch({ type: ActionTypes.OPPDATER_AVDOED, payload: { ...deepCopy(verdier), erValidert: false } })
        forrige!!()
    }

    const erValidert = state.omDenAvdoede.erValidert
    const selvstendigNaeringsdrivende = watch('selvstendigNaeringsdrivende.svar')
    const datoForDoedsfallet = watch('datoForDoedsfallet')

    return (
        <FormProvider {...methods}>
            <SkjemaElement>
                <Heading className={'center'} size={'medium'}>
                    {t('omDenAvdoede.tittel')}
                </Heading>
            </SkjemaElement>

            <form>
                <SkjemaGruppe>
                    <HGrid gap={'4'} columns={{ xs: 1, sm: 2 }} align={'start'}>
                        <RHFInput name={'fornavn'} label={t('omDenAvdoede.fornavn')} />
                        <RHFInput name={'etternavn'} label={t('omDenAvdoede.etternavn')} />

                        <RHFFoedselsnummerInput
                            name={'foedselsnummer'}
                            label={t('omDenAvdoede.foedselsnummer')}
                            description={t('felles.fnrPlaceholder')}
                        />
                        <RHFSelect
                            name={`statsborgerskap`}
                            label={t('omDenAvdoede.statsborgerskap')}
                            selectOptions={land}
                        />
                    </HGrid>

                    <SkjemaElement>
                        <Datovelger
                            name={'datoForDoedsfallet'}
                            label={t('omDenAvdoede.datoForDoedsfallet')}
                            maxDate={new Date()}
                        />
                    </SkjemaElement>

                    <SkjemaElement>
                        <RHFSpoersmaalRadio
                            name={'doedsfallAarsak'}
                            legend={t('omDenAvdoede.doedsfallAarsak')}
                            description={t('omDenAvdoede.doedsfallAarsakHvorfor')}
                            vetIkke
                        />
                    </SkjemaElement>
                </SkjemaGruppe>

                <BoddEllerArbeidetUtland datoForDoedsfallet={datoForDoedsfallet} />

                <SkjemaGruppe>
                    <Heading size="small">{t('omDenAvdoede.selvstendigNaeringsdrivende.tittel')}</Heading>
                    <BodyLong>{t('omDenAvdoede.selvstendigNaeringsdrivende.ingress')}</BodyLong>

                    <SkjemaElement>
                        <RHFSpoersmaalRadio
                            name={'selvstendigNaeringsdrivende.svar'}
                            legend={t('omDenAvdoede.selvstendigNaeringsdrivende.svar')}
                            vetIkke
                        />
                    </SkjemaElement>

                    {selvstendigNaeringsdrivende === IValg.JA && (
                        <>
                            <SkjemaElement>
                                <RHFValutaInput
                                    name={'selvstendigNaeringsdrivende.beskrivelse'}
                                    htmlSize={Bredde.S}
                                    valgfri={true}
                                    type="tel"
                                    description={t('omDenAvdoede.selvstendigNaeringsdrivende.placeholder')}
                                    label={t('omDenAvdoede.selvstendigNaeringsdrivende.beskrivelse')}
                                />
                            </SkjemaElement>

                            <RHFSpoersmaalRadio
                                name={'haddePensjonsgivendeInntekt.svar'}
                                legend={t('omDenAvdoede.haddePensjonsgivendeInntekt.svar')}
                                vetIkke
                            />
                        </>
                    )}
                </SkjemaGruppe>

                <Feilmeldinger errors={errors} />

                <Navigasjon
                    forrige={{ onClick: erValidert === true ? handleSubmit(lagreTilbake) : lagreTilbakeUtenValidering }}
                    neste={{ onClick: handleSubmit(lagreNeste) }}
                />
            </form>
        </FormProvider>
    )
}

export default OmDenAvdode
