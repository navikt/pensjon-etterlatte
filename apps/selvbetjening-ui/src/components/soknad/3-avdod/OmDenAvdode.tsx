import "../../../App.less";
import { SkjemaGruppe } from "nav-frontend-skjema";
import { Systemtittel } from "nav-frontend-typografi";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { IAvdoed } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { useTranslation } from "react-i18next";
import { FormProvider, useForm } from "react-hook-form";
import { Hovedknapp, Knapp } from "nav-frontend-knapper";
import { RHFFoedselsnummerInput, RHFInput } from "../../felles/RHFInput";
import { RHFToValgRadio } from "../../felles/RHFRadio";
import { IValg } from "../../../typer/Spoersmaal";
import Feilmeldinger from "../../felles/Feilmeldinger";

const OmDenAvdode: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const methods = useForm<IAvdoed>({
        defaultValues: state.omDenAvdoede || {},
        shouldUnregister: true
    });

    const {
        handleSubmit,
        watch,
        formState: { errors },
    } = methods;

    const lagre = (data: IAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_AVDOED, payload: data });
        neste!!();
    };

    const haddePensjonsgivendeInntekt = watch("haddePensjonsgivendeInntekt.svar")
    const mottokPensjonAndreLand = watch("mottokPensjonAndreLand.svar")
    const harAvtjentMilitaerTjeneste = watch("harAvtjentMilitaerTjeneste.svar")

    return (
        <FormProvider {...methods}>
            {/* Steg 3 */}
            <SkjemaGruppe>
                <Systemtittel className={"center"}>{t("omDenAvdoede.tittel")}</Systemtittel>
            </SkjemaGruppe>

            <form>
                {/* 3.1 */}
                <div className={"rad skjemagruppe"}>
                    <RHFInput
                        className={"kol-50"}
                        name={"fornavn"}
                        label={t("omDenAvdoede.fornavn")}
                    />

                    <RHFInput
                        className={"kol-50"}
                        name={"etternavn"}
                        label={t("omDenAvdoede.etternavn")}
                    />
                </div>

                <div className={"rad skjemagruppe"}>
                    {/* 3.2 */}
                    <RHFFoedselsnummerInput
                        className={"kol-50"}
                        type={"number"}
                        name={"foedselsnummer"}
                        label={t("omDenAvdoede.foedselsnummer")}
                    />

                    {/* 3.4 */}
                    <RHFInput
                        className={"kol-50"}
                        name={"statsborgerskap"}
                        label={t("omDenAvdoede.statsborgerskap")}
                    />
                </div>

                {/* 3.5 fjernes. Ikke lenger gyldig. */}
                {/* 3.6 */}
                <RHFToValgRadio
                    name={"bosetning"}
                    legend={t("omDenAvdoede.bosetning")}
                />

                {/* 3.7 */}
                <RHFToValgRadio
                    name={"doedsfallAarsak"}
                    legend={t("omDenAvdoede.doedsfallAarsak")}
                />

                {/* 3.8 */}
                <RHFToValgRadio
                    name={"boddEllerJobbetUtland"}
                    legend={t("omDenAvdoede.boddEllerJobbetUtland")}
                />
                {/* 3.9 Info om arbeidsforhold og inntekt hvis JA over */}

                {/* 3.10 */}
                <RHFToValgRadio
                    name={"haddePensjonsgivendeInntekt.svar"}
                    legend={t("omDenAvdoede.haddePensjonsgivendeInntekt.svar")}
                />

                {haddePensjonsgivendeInntekt === IValg.JA && (
                    <SkjemaGruppe>
                        <RHFInput
                            name={"haddePensjonsgivendeInntekt.beskrivelse"}
                            label={t("omDenAvdoede.haddePensjonsgivendeInntekt.beskrivelse")}
                        />
                    </SkjemaGruppe>
                )}
                {/* 3.11 Samme som over ?! */}

                {/* 3.12 */}
                <RHFToValgRadio
                    name={"mottokPensjonAndreLand.svar"}
                    legend={t("omDenAvdoede.mottokPensjonAndreLand.svar")}
                />

                {mottokPensjonAndreLand === IValg.JA && (
                    <SkjemaGruppe>
                        <RHFInput
                            name={"mottokPensjonAndreLand.beskrivelse"}
                            label={t("omDenAvdoede.mottokPensjonAndreLand.beskrivelse")}
                            rules={{ pattern: /^\d+$/ }}
                        />
                    </SkjemaGruppe>
                )}

                {/* 3.13 */}
                <RHFToValgRadio
                    name={"harAvtjentMilitaerTjeneste.svar"}
                    legend={t("omDenAvdoede.harAvtjentMilitaerTjeneste.svar")}
                />

                {harAvtjentMilitaerTjeneste === IValg.JA && (
                    <SkjemaGruppe>
                        <RHFInput
                            name={"harAvtjentMilitaerTjeneste.beskrivelse"}
                            label={t("omDenAvdoede.harAvtjentMilitaerTjeneste.beskrivelse")}
                            rules={{ pattern: /^\d{4}$/ }}
                        />
                    </SkjemaGruppe>
                )}

                <Feilmeldinger errors={errors}/>

                <SkjemaGruppe className={"navigasjon-rad"}>
                    <Knapp htmlType={"button"} onClick={forrige}>
                        {t("knapp.tilbake")}
                    </Knapp>

                    <Hovedknapp htmlType={"button"} onClick={handleSubmit(lagre)}>
                        {t("knapp.neste")}
                    </Hovedknapp>
                </SkjemaGruppe>
            </form>
        </FormProvider>
    );
};

export default OmDenAvdode;
