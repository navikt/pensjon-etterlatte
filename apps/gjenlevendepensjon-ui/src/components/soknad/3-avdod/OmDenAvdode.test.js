import { render } from "@testing-library/react";
import OmDenAvdode from "./OmDenAvdode";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    ...jest.requireActual("react-i18next"),
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe("Om den avdÃ¸de", () => {
    it("Snapshot", () => {
        const { container } = render(<OmDenAvdode />);
        expect(container).toMatchSnapshot();
    });
});
