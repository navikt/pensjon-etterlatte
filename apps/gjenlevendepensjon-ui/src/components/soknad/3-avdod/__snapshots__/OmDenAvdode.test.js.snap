// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om den avd√∏de Snapshot 1`] = `
<div>
  <fieldset
    aria-invalid="false"
    class="skjemagruppe center"
  >
    <h1
      class="navds-heading navds-heading--medium"
    >
      omDenAvdoede.tittel
    </h1>
    <div
      aria-live="polite"
    />
  </fieldset>
  <form>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe skjemagruppering"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe rad"
      >
        <div
          class="kolonne"
        >
          <p
            class="navds-label"
          >
            omDenAvdoede.fornavn
          </p>
          <p
            class="navds-body-long"
          />
        </div>
        <div
          class="kolonne"
        >
          <p
            class="navds-label"
          >
            omDenAvdoede.etternavn
          </p>
          <p
            class="navds-body-long"
          />
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
      <fieldset
        aria-invalid="false"
        class="skjemagruppe"
      >
        <div
          class="navds-grid"
          style="align-items: end;"
        >
          <div
            class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6 kol"
          >
            <div
              class="skjemaelement"
            >
              <label
                class="skjemaelement__label"
                for="foedselsnummer"
              >
                omDenAvdoede.foedselsnummer
              </label>
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="foedselsnummer"
                placeholder="felles.fnrPlaceholder"
                required=""
                type="tel"
                value=""
              />
            </div>
          </div>
          <div
            class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6 kol"
          >
            <div
              id="statsborgerskap"
            >
              <div
                class="skjemaelement"
              >
                <label
                  class="skjemaelement__label"
                >
                  omDenAvdoede.statsborgerskap
                </label>
                <div
                  class="selectContainer input--l"
                >
                  <select
                    aria-invalid="false"
                    class="skjemaelement__input"
                  >
                    <option
                      value=""
                    >
                      felles.velgLand
                    </option>
                  </select>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe skjemagruppering"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe ingress"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDenAvdoede.boddEllerJobbetUtland.tittel
        </h1>
        <p
          class="navds-body-long"
        >
          omDenAvdoede.boddEllerJobbetUtland.ingress
        </p>
        <div
          aria-live="polite"
        />
      </fieldset>
      <div
        id="boddEllerJobbetUtland.svar"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe inputPanelGruppe inline"
        >
          <legend
            class="skjemagruppe__legend"
          >
            omDenAvdoede.boddEllerJobbetUtland.svar
          </legend>
          <div
            class="inputPanelGruppe__inner"
          >
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="boddEllerJobbetUtland.svar"
                required=""
                type="radio"
                value="radiobuttons.ja"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.ja
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="boddEllerJobbetUtland.svar"
                required=""
                type="radio"
                value="radiobuttons.nei"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.nei
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="boddEllerJobbetUtland.svar"
                required=""
                type="radio"
                value="radiobuttons.vetIkke"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.vetIkke
              </span>
            </label>
          </div>
          <div
            aria-live="polite"
          />
        </fieldset>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe skjemagruppering"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe ingress"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDenAvdoede.selvstendigNaeringsdrivende.tittel
        </h1>
        <p
          class="navds-body-long"
        >
          omDenAvdoede.selvstendigNaeringsdrivende.ingress
        </p>
        <div
          aria-live="polite"
        />
      </fieldset>
      <div
        id="selvstendigNaeringsdrivende.svar"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe inputPanelGruppe inline"
        >
          <legend
            class="skjemagruppe__legend"
          >
            omDenAvdoede.selvstendigNaeringsdrivende.svar
          </legend>
          <div
            class="inputPanelGruppe__inner"
          >
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="selvstendigNaeringsdrivende.svar"
                required=""
                type="radio"
                value="radiobuttons.ja"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.ja
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="selvstendigNaeringsdrivende.svar"
                required=""
                type="radio"
                value="radiobuttons.nei"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.nei
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="selvstendigNaeringsdrivende.svar"
                required=""
                type="radio"
                value="radiobuttons.vetIkke"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.vetIkke
              </span>
            </label>
          </div>
          <div
            aria-live="polite"
          />
        </fieldset>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe skjemagruppering"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe ingress"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDenAvdoede.annenOpptjening.tittel
        </h1>
        <div
          aria-live="polite"
        />
      </fieldset>
      <fieldset
        aria-invalid="false"
        class="skjemagruppe"
      >
        <div
          id="doedsfallAarsak"
        >
          <fieldset
            aria-invalid="false"
            class="skjemagruppe inputPanelGruppe inline"
          >
            <legend
              class="skjemagruppe__legend"
            >
              omDenAvdoede.doedsfallAarsak
            </legend>
            <div
              class="skjemagruppe__description"
            >
              <div
                class="hvorforPanel"
              >
                <button
                  aria-expanded="false"
                  class="hvorforPanel__toggle"
                  data-testid="hvorforPanel_toggle"
                  type="button"
                >
                  <span>
                    hvorforSpoerVi
                  </span>
                  <span>
                    <svg
                      fill="none"
                      height="1em"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12 17L3 8.429 4.5 7l7.5 7.143L19.5 7 21 8.429 12 17z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="inputPanelGruppe__inner"
            >
              <label
                class="inputPanel radioPanel"
              >
                <input
                  aria-checked="false"
                  aria-invalid="false"
                  class="inputPanel__field"
                  name="doedsfallAarsak"
                  required=""
                  type="radio"
                  value="radiobuttons.ja"
                />
                <span
                  class="inputPanel__label"
                >
                  radiobuttons.ja
                </span>
              </label>
              <label
                class="inputPanel radioPanel"
              >
                <input
                  aria-checked="false"
                  aria-invalid="false"
                  class="inputPanel__field"
                  name="doedsfallAarsak"
                  required=""
                  type="radio"
                  value="radiobuttons.nei"
                />
                <span
                  class="inputPanel__label"
                >
                  radiobuttons.nei
                </span>
              </label>
              <label
                class="inputPanel radioPanel"
              >
                <input
                  aria-checked="false"
                  aria-invalid="false"
                  class="inputPanel__field"
                  name="doedsfallAarsak"
                  required=""
                  type="radio"
                  value="radiobuttons.vetIkke"
                />
                <span
                  class="inputPanel__label"
                >
                  radiobuttons.vetIkke
                </span>
              </label>
            </div>
            <div
              aria-live="polite"
            />
          </fieldset>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
      <div
        id="harAvtjentMilitaerTjeneste.svar"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe inputPanelGruppe inline"
        >
          <legend
            class="skjemagruppe__legend"
          >
            omDenAvdoede.harAvtjentMilitaerTjeneste.svar
          </legend>
          <div
            class="skjemagruppe__description"
          >
            <div
              class="hvorforPanel"
            >
              <button
                aria-expanded="false"
                class="hvorforPanel__toggle"
                data-testid="hvorforPanel_toggle"
                type="button"
              >
                <span>
                  hvorforSpoerVi
                </span>
                <span>
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 17L3 8.429 4.5 7l7.5 7.143L19.5 7 21 8.429 12 17z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="inputPanelGruppe__inner"
          >
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="harAvtjentMilitaerTjeneste.svar"
                required=""
                type="radio"
                value="radiobuttons.ja"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.ja
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="harAvtjentMilitaerTjeneste.svar"
                required=""
                type="radio"
                value="radiobuttons.nei"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.nei
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="harAvtjentMilitaerTjeneste.svar"
                required=""
                type="radio"
                value="radiobuttons.vetIkke"
              />
              <span
                class="inputPanel__label"
              >
                radiobuttons.vetIkke
              </span>
            </label>
          </div>
          <div
            aria-live="polite"
          />
        </fieldset>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-footer"
    >
      <div
        class="navigasjon-rad"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.tilbake
          </span>
        </button>
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.neste
          </span>
        </button>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-rad"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        id="avbryt-btn"
        type="button"
      >
        <span
          class="navds-button__inner navds-body-short"
        >
          knapp.avbryt
        </span>
      </button>
      <div
        aria-live="polite"
      />
    </fieldset>
  </form>
</div>
`;
